client/src/components/AuthForm/LogInForm/LogInForm.js

```diff
@@ -27,13 +36,16 @@ const LogInForm = ({ login, isAuthenticated, loading }) => {
     setLoginForm({ ...logInForm, [e.target.name]: e.target.value });
   };
 
-  const handleSubmit = async (e) => {
-    try {
-      e.preventDefault();
-      login(logInForm);
-    } catch {
-      // @TODO: make error appear as a jsx element in error colors
-      throw new Error('Failed to log in');
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (!validator.isEmail(logInForm.email)) {
+      dispatch(setAlert('Enter a valid email address.', 'Error'));
+    } else if (logInForm.password.length < 6 || logInForm.password === '') {
```

Comment:
else if (logInForm.password.length < 6 || logInForm.password === '')

should be refactored as

 else if (logInForm.password.length < 6 || !logInForm.password)
because an empty string is falsy, the === '' is redundant